modules/hal_hpmicro/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# Copyright 2022 hpmicro
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_HAS_HPMICRO_HAL)
+
+set(HPM_SDK_BASE ${ZEPHYR_HAL_HPMICRO_MODULE_DIR}/hpm_sdk)
+set(HPM_BOARD_DIR ${CONFIG_BOARD})
+string(TOUPPER ${SOC_NAME} HPM_SOC_TOUPPER)
+set(HPM_SOC ${HPM_SOC_TOUPPER})
```

Comment:
please indicate where are these used.

Reply:
A module named `hal_hpmicro` has been added, which internally relies on these macros to control the behaviour of cmake. This format is good for code maintenance, if `hpm_sdk` is updated it can be copied over directly without having to change anything.

Reply:
why isn't this done at HAL level then? or all in `modules/`? It feels like CMake is splitted in 2, making it hard to know what's going on here.

Reply:
The cmake for hal_hpmicro is the cmake for hpm_sdk, which requires some parameters to be passed in by the user when using it. After being integrated into zephyr, we want to be able to maintain compatibility with hpm_sdk without having to maintain an additional environment.